crates/matrix-sdk-base/src/client.rs

```diff
@@ -561,24 +564,86 @@ impl BaseClient {
         changed_devices: &api::sync::sync_events::DeviceLists,
         one_time_keys_counts: &BTreeMap<ruma::DeviceKeyAlgorithm, UInt>,
         unused_fallback_keys: Option<&[ruma::DeviceKeyAlgorithm]>,
+        changes: &mut StateChanges,
     ) -> Result<Vec<Raw<ruma::events::AnyToDeviceEvent>>> {
         if let Some(o) = self.olm_machine().await.as_ref() {
             // Let the crypto machine handle the sync response, this
             // decrypts to-device events, but leaves room events alone.
             // This makes sure that we have the decryption keys for the room
             // events at hand.
-            Ok(o.receive_sync_changes(
-                to_device_events,
-                changed_devices,
-                one_time_keys_counts,
-                unused_fallback_keys,
-            )
-            .await?)
+            let (events, room_key_updates) = o
+                .receive_sync_changes(
+                    to_device_events,
+                    changed_devices,
+                    one_time_keys_counts,
+                    unused_fallback_keys,
+                )
+                .await?;
+
+            #[cfg(feature = "experimental-sliding-sync")]
+            for room_key_update in room_key_updates {
+                if let Some(mut room) = self.get_room(&room_key_update.room_id) {
+                    self.decrypt_latest_events(&mut room, changes).await;
+                }
+            }
```

Comment:
How is this sliding-sync specific?

Reply:
It uses `decrypt_latest_events`, which uses `latest_event` and `latest_encrypted_events` which are both sliding-sync specific (for now)